cobaya/model.py

```diff
@@ -399,39 +528,35 @@ def logposterior(self, params_values, return_derived=True,
                     self.log, "Got non-finite parameter values: %r",
                     dict(zip(self.parameterization.sampled_params(),
                              params_values_array)))
-
         # Notice that we don't use the make_finite in the prior call,
         # to correctly check if we have to compute the likelihood
-        logps = self.prior.logps_internal(params_values_array)
-        if logps == -np.inf:
+        logpriors_1d = self.prior.logps_internal(params_values_array)
+        if logpriors_1d == -np.inf:
             logpriors = [-np.inf] * (1 + len(self.prior.external))
-            logpost = -np.inf
         else:
             input_params = self.parameterization.to_input(params_values_array)
-            logpriors = [logps]
+            logpriors = [logpriors_1d]
             if self.prior.external:
                 logpriors.extend(self.prior.logps_external(input_params))
-                logpost = sum(logpriors)
-            else:
-                logpost = logps
-
-        if logps != -np.inf:
+        if logpriors_1d != -np.inf:
```

Comment:
This check (both before and after the PR) goes on to compute the likelihoods even if "external" priors are null. I don't think it should be that way: sometimes poeple use external priors to exclude large parameter regions, and I would prefer not to waste time computing the likelihood there. @cmbant if you think this behaviour is useful, maybe we can make it optional?

Reply:
Probably just an oversight, fine my me to change.